source/DSCResources/DSC_PostgreSqlScript/DSC_PostgreSqlScript.psm1

```diff
@@ -0,0 +1,268 @@
+                & $PsqlLocation -c "CREATE DATABASE $DatabaseName"
+            }
+        }
+        Invoke-Command -ScriptBlock {
+            & $PsqlLocation -d $DatabaseName -f $SetFilePath 2>&1
+        }
+    }
+    catch [System.Management.Automation.CommandNotFoundException]
+    {
+        Write-Verbose -Message ($script:localizedData.PsqlNotFound -f $PsqlLocation)
+        throw $_
+    }
+    catch
+    {
+        throw $_
+    }
+    finally
+    {
+        $ErrorActionPreference = $previousErrorActionPreference
+    }
+}
+
+<#
+    .SYNOPSIS
+        Evaluates the value returned from the Test-script.
+    .PARAMETER DatabaseName
+        Specifies the name of the PostgreSQL database.
+    .PARAMETER SetFilePath
+        Path to the T-SQL file that will perform Set action.
+    .PARAMETER GetFilePath
+        Path to the T-SQL file that will perform Get action.
+        Any values returned by the T-SQL queries will also be returned by the cmdlet Get-DscConfiguration through the `GetResult` property.
+    .PARAMETER TestFilePath
+        Path to the T-SQL file that will perform Test action.
+        Any script that does not throw an error or returns null is evaluated to true.
+        The cmdlet Invoke-Sqlcmd treats T-SQL Print statements as verbose text, and will not cause the test to return false.
+    .PARAMETER Credential
+        The credentials to authenticate with, using PostgreSQL Authentication.
+    .PARAMETER PsqlLocation
+        Location of the psql executable.  Defaults to "C:\Program Files\PostgreSQL\12\bin\psql.exe".
+#>
+function Test-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Boolean])]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $DatabaseName,
+
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $SetFilePath,
+
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $GetFilePath,
+
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $TestFilePath,
+
+        [Parameter(Mandatory = $true)]
+        [System.Management.Automation.PSCredential]
+        [System.Management.Automation.Credential()]
+        $Credential,
+
+        [Parameter()]
+        [System.String]
+        $PsqlLocation = "C:\Program Files\PostgreSQL\12\bin\psql.exe"
+    )
+
+    $env:PGPASSWORD = $Credential.Password.ToString()
+    $env:PGUSER = $Credential.UserName
+
+    try
+    {
+        Write-Verbose -Message ($script:localizedData.ExecutingTestScript -f $TestFilePath,$DatabaseName)
+        # Must redirect error stream into output stream to capture some errors from psql
+        $previousErrorActionPreference = $ErrorActionPreference
+        $ErrorActionPreference = "Stop"
+        $result = Invoke-Command -ScriptBlock {
+            & $PsqlLocation -d $DatabaseName -f $TestFilePath 2>&1
+        }
+
+        if ($result -eq [bool]::TrueString)
+        {
+            return $true
```

Comment:
Usually we wouldn't return here but set to a variable and then return that at the end and Write-Verbose at the end stating what that state is that is being returned.